src/closed.jl

```diff
@@ -0,0 +1,44 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
+"""
+immutable ClosedSet{T}
+	left::T
+	right::T
+end
+
+function ClosedSet(left, right)
+	(I_left, I_right) = promote(left, right)
+	ClosedSet(I_left, I_right)
+end
+
+show(io::IO, I::ClosedSet) = print(io, "[", I.left, ",", I.right, "]")
+
+length(I::ClosedSet) = I.right - I.left + 1
+
+in(v, I::ClosedSet) = I.left <= v <= I.right
+
+isempty(A::ClosedSet) = A.left > A.right
+
+isequal(A::ClosedSet, B::ClosedSet) = (A.left == B.left && A.right == B.right) || isempty(A) && isempty(B)
+
+==(A::ClosedSet, B::ClosedSet) = isequal(A, B)
```

Comment:
Hmmm, interesting. Not entirely sure, but I wonder if `0.0 == -0.0` is justification for saying that all empty sets should return `true` for `==`.

It seems kind of odd that `==` is more permissive than `isequal` about the sign of 0 but less permissive about `NaN`.